components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/JWTTokenSecurityHandler.java

```diff
@@ -0,0 +1,153 @@
+                //Uses in memory user store
+                return processLoginRequestInMemoryUserStore(authHeaderToken);
+            }
+        } else { //Other resources apart from /login should be authenticated from JWT based auth
+            JWTTokenStore tokenStore = JWTInMemoryTokenStore.getInstance();
+            tokenStore.removeExpired();
+            JWTTokenInfoDTO jwtTokenInfoDTO = tokenStore.getToken(authHeaderToken);
+            if (jwtTokenInfoDTO != null && !jwtTokenInfoDTO.isRevoked()) {
+                jwtTokenInfoDTO.setLastAccess(System.currentTimeMillis()); //Record last successful access
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if a given value is not null and not empty.
+     *
+     * @param value String value
+     */
+    private Boolean isValid(String value) {
+        return (Objects.nonNull(value) && !value.isEmpty());
+    }
+
+    /**
+     * Processes /login request if the JWTToken Security Handler is engaged. Since /login is
+     * basic auth
+     * @param token extracted basic auth token
+     * @return
+     */
+    private boolean processLoginRequestInMemoryUserStore(String token) {
+        String decodedCredentials = new String(new Base64().decode(token.getBytes()));
+        String[] usernamePasswordArray = decodedCredentials.split(":");
+        if(userList == null || userList.isEmpty()) {
+            populateUserList();
+        }
+        if (usernamePasswordArray.length != 2) {
+            return false;
+        }
+        String username = usernamePasswordArray[0];
+        String password = usernamePasswordArray[1];
+        if (!userList.isEmpty()) {
+            for (String userNameFromStore : userList.keySet()) {
+                if (userNameFromStore.equals(username)) {
+                    String passwordFromStore = String.valueOf(userList.get(userNameFromStore));
+                    if (isValid(passwordFromStore) && passwordFromStore.equals(password)) {
+                        LOG.info("User " + username + " logged in successfully");
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Processes /login request if the JWTToken Security Handler is engaged. Since /login is
+     * basic auth
+     * @param token extracted basic auth token
+     * @return
+     */
+    private boolean processLoginRequestWithCarbonUserStore(String token) throws UserStoreException {
+        String decodedCredentials = new String(new Base64().decode(token.getBytes()));
+        String[] usernamePasswordArray = decodedCredentials.split(":");
+        if (usernamePasswordArray.length != 2) {
+            return false;
+        }
+        String username = usernamePasswordArray[0];
+        String password = usernamePasswordArray[1];
+        boolean isAuthenticated = DataHolder.getInstance().getUserStoreManager().authenticate(username,
```

Comment:
Let's use org.wso2.micro.integrator.security.MicroIntegratorSecurityUtils#getUserStoreManager to retrieve user store manager because there is a lazy loading option for user store

Reply:
Thanks fixed